src/VivViewer.js

```diff
@@ -2,43 +2,155 @@ import React, { PureComponent } from 'react';
+      const { imageWidth, imageHeight } = loader.getRasterSize({
+        z: 0
+      });
+      const newViewState = {};
+      newViewState.detail = viewState;
+      newViewState.overview = {
+        ...viewState,
+        target: [imageWidth / 2, imageHeight / 2, 0],
+        zoom: -(numLevels - 1)
+      };
+      this.setState({ viewState: newViewState });
+    }
+  }
+
+  // For now this is hardcoded but in general we should look at
+  // a proper structure for taking lists of configurations so that
+  // we can handle multiple overlapping layers.
+  // https://github.com/hubmapconsortium/vitessce-image-viewer/issues/107
   _renderLayers() {
-    const { loader } = this.props;
-    // For now this is hardcoded but in general we should look at
-    // a proper structure for taking lists of configurations so that
-    // we can handle multiple overlapping layers.
-    // https://github.com/hubmapconsortium/vitessce-image-viewer/issues/107
-    return loader.isPyramid
-      ? new VivViewerLayer({
-          id: `VivViewerLayer-${loader.type}`,
+    const { loader, overviewOn } = this.props;
+    if (loader.isPyramid) {
+      const layers = [
+        new VivViewerLayer({
+          id: `${loader.type}-detail`,
+          // Because TileLayer is unique in updating on viewport changes,
+          // it needs to be aware of what viewport it is rendering in -
+          // layerFilter only handles `draw` calls.
+          viewportId: 'detail',
           ...this.props
         })
-      : new StaticImageLayer({
-          id: `StaticImageLayer-${loader.type}`,
-          ...this.props
+      ];
+      if (overviewOn) {
+        const { viewHeight, viewWidth } = this.props;
+        const { viewState } = this.state;
+        const viewport = new OrthographicView().makeViewport({
+          // From the current `detail` viewState, we need its projection matrix (actually the inverse).
+          viewState: viewState.detail,
+          height: viewHeight,
+          width: viewWidth
         });
+        // Use the inverse of the projection matrix to map screen to the view space.
+        const boundingBox = [
+          viewport.unproject([0, 0]),
+          viewport.unproject([viewport.width, 0]),
+          viewport.unproject([viewport.width, viewport.height]),
+          viewport.unproject([0, viewport.height])
+        ];
+        layers.push(
+          new OverviewLayer(this.props, {
+            id: `${loader.type}-overview`,
+            boundingBox,
+            ...this.props
+          })
+        );
+        return layers;
+      }
+      return layers;
+    }
+
+    return new StaticImageLayer({
+      id: `StaticImageLayer-${loader.type}-detail`,
+      ...this.props
+    });
   }
 
   render() {
     /* eslint-disable react/destructuring-assignment */
+    const { loader, overviewOn } = this.props;
+    const { numLevels } = loader;
+    const { imageWidth, imageHeight } = loader.getRasterSize({
+      z: 0
+    });
```

Comment:
My rationale around changing the return type for `getRaster` was to make anyone else implementing a loader avoid the need to implement another class method. But, it might make sense to change the return of `getRaster` back to an array of TypedArrays if we are going to have this method anyways...

It will make the returns between `getTile` and `getRaster` the same

Reply:
I leave that up to you.  I think it's fair to expect `getRaster` to return some extra information since that information is not really inferrable in any way, unlike `getTileData` which is always the same.  If you want me to, I can change it back to arrays with `getRaster` - let me know.

Reply:
BTW I was mostly playing devil's advocate there.  `geotiff` does this where the return additional data.  I don't have a strong feeling here - I think we should have a "yes" or "no" to either way.  I mentioned the need for this in passing a while ago but I should have been clearer about it.

Reply:
FWIW I think we should have `getTile` return an object as well, so that the returns are similar. I notice that `image.getTileOrStrip` returns a data object as well, and so does `z.getRawChunk`. I think it would be fair to return:

```
{ data, width, height }
```

Where width and height for `getTile` would be `tileSize` and `tileSize`. We don't require this in viv, but I think it would this would hopefully make the loaders more generally useful.

Reply:
we would just need to wrap:
```javascript
const getTileData =  async ({ x, y, z }) => { 
   const { data } = await loader.getTile({x, y, z: -z});
   return data
}
```